core/src/main/scala/org/apache/spark/sql/SnappyParser.scala

```diff
@@ -499,6 +499,12 @@ class SnappyParser(session: SnappySession)
   }
 
   protected final def relationFactor: Rule1[LogicalPlan] = rule {
+    identifier ~ ws ~ '(' ~ ws ~ (expression * commaSep) ~ ws ~ ')' ~>
+        ((fnname: String, e: Any) =>
+          e match {
+            case ve: Vector[Expression] => UnresolvedTableValuedFunction(fnname.toLowerCase(), ve)
+            case _ => UnresolvedTableValuedFunction(fnname, Seq.empty)
+          }) |
```

Comment:
Don't think we expect it to be the most common case, so better to move this case below the others rather than being the first rule. Note that parser will backtrack when it fails to match the rule, then go to other rules so the most common rules should be at the start.

In fact best option will be to factor out identifier which is common in this rule and tableIdentifier as has been done in many other rules like comparisonExpression. Can leave that marked for TODO for later but moving this rule at the end seems to be the better option for now.